File: code_challenge_solution/c#/86 Hangman.cs
Here is the C# code for creating a Hangman word guessing game:

```csharp
using System;

namespace HangmanGame
{
    class Program
    {
        static void Main(string[] args)
        {
            string secretWord = "hangman"; // The secret word to be guessed
            string guessedWord = ""; // The word that the user is trying to guess
            int attempts = 7; // Number of attempts remaining

            Console.WriteLine("Welcome to Hangman!");

            // Populate the guessedWord with underscores
            for (int i = 0; i < secretWord.Length; i++)
            {
                guessedWord += "_";
            }

            while (attempts > 0)
            {
                Console.WriteLine("\n" + guessedWord);
                Console.WriteLine("Attempts remaining: " + attempts);
                Console.Write("Enter a letter: ");
                char guess = Console.ReadLine()[0];

                bool found = false;

                // Check if the guessed letter is present in the secret word
                for (int i = 0; i < secretWord.Length; i++)
                {
                    if (secretWord[i] == guess)
                    {
                        // Replace the corresponding underscore in guessedWord with the guessed letter
                        guessedWord = guessedWord.Substring(0, i) + guess + guessedWord.Substring(i + 1);
                        found = true;
                    }
                }

                if (found)
                {
                    Console.WriteLine("Correct!");
                }
                else
                {
                    Console.WriteLine("Wrong!");
                    attempts--;
                }

                // Check if the word has been fully guessed
                if (guessedWord == secretWord)
                {
                    Console.WriteLine("\nCongratulations! You guessed the word: " + secretWord);
                    break;
                }
            }

            if (attempts <= 0)
            {
                Console.WriteLine("\nGame Over! The word was: " + secretWord);
            }

            Console.WriteLine("\nPress any key to exit.");
            Console.ReadKey();
        }
    }
}
```

File: code_challenge_solution/c++/86 Hangman.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

void printHangman(int hangmanState) {
    if(hangmanState == 0) {
        cout << " _________     " << endl;
        cout << "|         |    " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
    } else if(hangmanState == 1) {
        cout << " _________     " << endl;
        cout << "|         |    " << endl;
        cout << "|         0    " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
    } else if(hangmanState == 2) {
        cout << " _________     " << endl;
        cout << "|         |    " << endl;
        cout << "|         0    " << endl;
        cout << "|         |    " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
    } else if(hangmanState == 3) {
        cout << " _________     " << endl;
        cout << "|         |    " << endl;
        cout << "|         0    " << endl;
        cout << "|        /|    " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
    } else if(hangmanState == 4) {
        cout << " _________     " << endl;
        cout << "|         |    " << endl;
        cout << "|         0    " << endl;
        cout << "|        /|\\  " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
    } else if(hangmanState == 5) {
        cout << " _________     " << endl;
        cout << "|         |    " << endl;
        cout << "|         0    " << endl;
        cout << "|        /|\\  " << endl;
        cout << "|        /     " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
    } else {
        cout << " _________     " << endl;
        cout << "|         |    " << endl;
        cout << "|         0    " << endl;
        cout << "|        /|\\  " << endl;
        cout << "|        / \\  " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
        cout << "|              " << endl;
    }
}

bool isWordGuessed(string word, vector<char> guesses) {
    for(int i = 0; i < word.length(); i++) {
        bool guessed = false;
        for(char c : guesses) {
            if(word[i] == c) {
                guessed = true;
                break;
            }
        }
        if(!guessed) {
            return false;
        }
    }
    return true;
}

void printWord(string word, vector<char> guesses) {
    for(int i = 0; i < word.length(); i++) {
        bool guessed = false;
        for(char c : guesses) {
            if(word[i] == c) {
                guessed = true;
                break;
            }
        }
        if(guessed) {
            cout << word[i] << " ";
        } else {
            cout << "_ ";
        }
    }
    cout << endl;
}

int main() {
    vector<string> wordList = {"apple", "banana", "cherry", "durian", "elderberry", "fig"};
    srand(time(0));
    int randomIndex = rand() % wordList.size();
    string wordToGuess = wordList[randomIndex];

    int attempts = 0;
    vector<char> guesses;

    cout << "Welcome to Hangman!" << endl;

    while(attempts < 6) {
        cout << endl;
        printHangman(attempts);
        cout << endl;
        printWord(wordToGuess, guesses);
        cout << endl;

        cout << "Enter a letter: ";
        char guess;
        cin >> guess;

        guesses.push_back(guess);

        if(wordToGuess.find(guess) != string::npos) {
            cout << "Correct guess!" << endl;
        } else {
            cout << "Wrong guess!" << endl;
            attempts++;
        }

        if(isWordGuessed(wordToGuess, guesses)) {
            cout << endl;
            printWord(wordToGuess, guesses);
            cout << endl;
            cout << "Congratulations! You won!" << endl;
            break;
        }
    }

    if(attempts == 6) {
        cout << "You lost. The word was " << wordToGuess << "." << endl;
    }

    return 0;
}